tests/common_setup.py

```diff
@@ -139,20 +139,21 @@ def standard_setup_with_short_lived_token():
 
 @pytest.fixture(scope="function")
 def running_custom_production_setup(request):
-    conftest.production_setup_lock.acquire()
+    with conftest.production_setup_lock.acquire():
+        # since we are starting a manual instance of the backend,
+        # let the script know the instance is called "testprod"
+        # so that is cleaned up correctly on test failure/error
 
-    # since we are starting a manual instance of the backend,
-    # let the script know the instance is called "testprod"
-    # so that is cleaned up correctly on test failure/error
+        def fin():
+            stop_docker_compose()
+            conftest.production_setup_lock.release()
 
-    def fin():
-        conftest.production_setup_lock.release()
-        stop_docker_compose()
+        # this additional lock is required
+        conftest.production_setup_lock.acquire()
```

Comment:
This looks wrong, you're acquiring the lock twice here, once using `with`, and once here. When do you want to hold the lock, during the whole production environment being up, or only during initialization and teardown?